pkg/cmd/release-payload-controller/payload_verification_controller.go

```diff
@@ -0,0 +1,188 @@
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid queue key '%v': %v", obj, err))
+		return
+	}
+	c.queue.Add(key)
+}
+
+func (c *PayloadVerificationController) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+
+	klog.Info("Starting Payload Verification Controller")
+	defer func() {
+		klog.Info("Shutting down Payload Verification Controller")
+		c.queue.ShutDown()
+		klog.Info("Payload Verification Controller shut down")
+	}()
+
+	if !cache.WaitForNamedCacheSync("PayloadVerificationController", ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go func() {
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *PayloadVerificationController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *PayloadVerificationController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx, key.(string))
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *PayloadVerificationController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting PayloadVerificationController sync")
+	defer klog.V(4).Infof("PayloadVerificationController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	// The ReleasePayload resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	releasePayload := originalReleasePayload.DeepCopy()
+
+	// If there are any JobResults defined, then we don't need to do anything else here...
+	if len(releasePayload.Status.BlockingJobsResults) != 0 || len(releasePayload.Status.InformingJobsResults) != 0 {
+		klog.V(5).Infof("ReleasePayload: '%s/%s' already synced", releasePayload.Namespace, releasePayload.Name)
+		return nil
+	}
+
+	klog.V(4).Infof("Syncing Payload Verification for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	for _, verify := range releasePayload.Spec.PayloadVerificationConfig.BlockingJobs {
+		releasePayload.Status.BlockingJobsResults = append(releasePayload.Status.BlockingJobsResults, generateJobStatus(verify))
+	}
+
+	for _, verify := range releasePayload.Spec.PayloadVerificationConfig.InformingJobs {
+		releasePayload.Status.InformingJobsResults = append(releasePayload.Status.InformingJobsResults, generateJobStatus(verify))
+	}
+
+	if !reflect.DeepEqual(originalReleasePayload, releasePayload) {
```

Comment:
you need to canonicalize your release payload.  So that you always present status in the same order.  Imagine what happens if the array is say non-alphabetized.

Reply:
Done!